src/test/java/com/aws/greengrass/mqttclient/MqttClientTest.java

```diff
@@ -101,6 +103,7 @@ void beforeEach() {
         lenient().when(mockConnection.unsubscribe(any())).thenReturn(CompletableFuture.completedFuture(0));
         lenient().when(mockConnection.publish(any(), any(), anyBoolean()))
                 .thenReturn(CompletableFuture.completedFuture(0));
+        Thread.interrupted();
```

Comment:
why are you interrupting the main thread?

Reply:
This is for the testing issue. In order to make it method exit the infinite loop, I introduced the InterruptedException and called Thread.currentThread().interrupt();  when exiting the loop. This is to clear the interrupting flag.  https://github.com/junit-team/junit4/issues/1365

Reply:
But this line should be removed, it should be only applied to a specific test.

Reply:
You're running `publishAllTasks` on the main thread? Why is the thread getting interrupted then, who is doing the interrupting?